downloader.py

```diff
@@ -25,91 +30,150 @@
 import argparse
 import glob
 
-# Default values
-create_folder = True
-create_plst = False
-
-# CMD arguments
-parser = argparse.ArgumentParser(
-    description='Download mp3 files from provided youtube playlist URL'
-)
-parser.add_argument(
-    '-u',
-    '--url',
-    action='store',
-    dest='url',
-    default=None,
-    help='Required youtube URL. You can pass either playlist or single video URL',
-    required=True
-)
-parser.add_argument(
-    '-f',
-    '--not-create-folder',
-    action='store_false',
-    dest='fold',
-    help='Don\'t create a folder'
-)
-parser.add_argument(
-    '-p',
-    '--playlist',
-    action='store_true',
-    dest='plst',
-    help='Create an m3u playlist from downloaded files'
-)
-
-parser.set_defaults(fold=True, plst=False)
-results = parser.parse_args()
-url = results.url
-create_folder = results.fold
-create_plst = results.plst
-
-# Get playlist title and create a folder for it the argument
-# -f wasn't present
-foldername = ''
-if create_folder:
-    ydl_init_opts = {
-        'quiet': True,
-        'extract_flat': True
+
+def dl_audio(url, folder=True, create_plst=False, download_plst=True):
+    """Download audio from provided url and save it to hard drive."""
+    # Get playlist title and create a folder for it the argument
+    # -f wasn't present
+    foldername = ''
+    if folder:
+        if isinstance(url, list):
+            # the last item in the list is filename
+            # fodername is filename without extension
+            foldername = '{}/'.format(str(url[-1][:-4]))
+        else:
+            ydl_init_opts = {
+                'quiet': True,
+                'extract_flat': True
+            }
+            ydl_init = YoutubeDL(ydl_init_opts)
+            ydl_init.add_default_info_extractors()
+            info = ydl_init.extract_info(url, download=False)
+
+            foldername = '{}/'.format(info['title'])
+
+        if not os.path.exists(foldername):
+            os.makedirs(foldername)
+
+    # Options for actual audio download
+
+    if download_playlist:
+        plst_opt = 'yesplaylist'
+    else:
+        plst_opt = 'noplaylist'
+    ydl_opts = {
+        'verbose': True,
+        'outtmpl': '{}%(title)s.%(ext)s'.format(foldername),
+        'download_archive': '{}downloads.txt'.format(foldername),
+        'format': 'bestaudio/best',  # choice of quality
+        plst_opt: True,        # download playlist or not
+        'ignoreerrors': True,       # ignore errors
```

Comment:
I think this comment (`ignore errors`) is not needed, as the option itself is quite self-explanatory.